lib/api.js

```diff
@@ -115,9 +130,20 @@ function makeApiRequest(transport, opts, bodyObj, callback) {
     });
   });
 
+  var originalArguments = arguments;
   req.on('error', function (err) {
-    logger.error('Could not make request to rollbar, ' + err);
-    callback(err);
+    // If the request to Rollbar failed due to a connection error, lets queue
+    // up the requests and try again periodically in the hopes that the connection
+    // will be restored.
+    //
+    // If no retryInterval set, interpret that as not wanting to retry send errors,
+    // so just bail out silently.
+    if (SETTINGS.retryInterval && err.code == 'ENOTFOUND') {
```

Comment:
I haven't been able to find a comprehensive list of error codes that are sent back from an HTTP request.  I believe these are just bubbled up from the underlying socket libraries.  One option is to just retry on _any_ error.  That actually may be the better option.  Thoughts?